From 4366a409b5e30a53dbd4fa91bfcb9b31de0bb39e Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Tue, 25 Oct 2011 17:27:02 +0200
Subject: [PATCH] savevm: qemu_fille_buffer() used to return one error for
 reads of size 0.

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <e84f81ec44707e1e0a981eac5603a42e53c0491b.1319563341.git.quintela@redhat.com>
Patchwork-id: 34642
O-Subject: [PATCH] savevm: qemu_fille_buffer() used to return one error for reads of size 0.
Bugzilla: 669581
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>

Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 savevm.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 savevm.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/savevm.c b/savevm.c
index c995d28..db43dab 100644
--- a/savevm.c
+++ b/savevm.c
@@ -475,6 +475,8 @@ static void qemu_fill_buffer(QEMUFile *f)
     if (len > 0) {
         f->buf_size += len;
         f->buf_offset += len;
+    } else if (len == 0) {
+        f->last_error = -EIO;
     } else if (len != -EAGAIN)
         f->last_error = len;
 }
-- 
1.7.4.4

